ONSdigital/babbage
# Fix chart source overlap

### What

Fix the bug where the source will overlap on long legends:
<img src="https://trello-attachments.s3.amazonaws.com/599ebbc8b447bdb9df2b7229/59cb60fae927123f17669987/x/bce3f7f4b504442e8b87b1cb10950dfc/image.png"/>

The source text is now being rendered by Babbage, rather than by the Highcharts server - meaning we have more control over spacing. We couldn't fix it with Highcharts itself because the legend can be any size and we can't add the source below it dynamically or know the height of the legend. This means we render the chart image using Highcharts and then put the source onto the bottom of the image ourselves.

### How to review

A code review is needed, but to check frontend functionality you need to:
1) Have a chart on your local machine and be on the branch `fix/chart-source-overlap` on Babbage.
2) View the non-js version of the page to check the source isn't in the chart image.
4) Download the image, using the button under the heading 'Download this chart'.
5) Check that the downloaded image has a source on it.
6) Ignore other bugs (not related to the source) such as too much spacing around the chart, the title being centred/duplicated etc. There are plenty of other things to fix but not as part of this update!

### Who can review

@daiLlew please.
